protostar/commands/deploy/deploy_command.py

```diff
@@ -31,19 +52,78 @@ def arguments(self) -> List[Command.Argument]:
                 is_array=True,
             ),
             Command.Argument(
-                name="network",
-                description="A name of the network defined in protostar.toml",
+                name="token",
+                description="Used for deploying contracts in Alpha MainNet.",
                 type="str",
             ),
             Command.Argument(
-                name="token",
-                description="Used for deploying contracts in Alpha MainNet.",
+                name="salt",
+                description=(
+                    "An optional salt controlling where the contract will be deployed. "
+                    "The contract deployment address is determined by the hash "
+                    "of contract, salt and caller. "
+                    "If the salt is not supplied, the contract will be deployed with a random salt."
+                ),
                 type="str",
             ),
+            self._gateway_url_arg,
+            self._network_arg,
         ]
 
-    async def run(self, _args):
-        return await self.deploy()
+    async def run(self, args):
+        return await self.deploy(
+            contract_name=args.contract,
+            network=args.network,
+            inputs=args.inputs,
+            token=args.token,
+            salt=args.salt,
+        )
+
+    # pylint: disable=too-many-arguments
+    async def deploy(
+        self,
+        contract_name: str,
+        inputs: Optional[List[str]] = None,
+        network: Optional[str] = None,
+        gateway_url: Optional[str] = None,
+        token: Optional[str] = None,
+        salt: Optional[str] = None,
+    ):
+        with open(
+            self._project.project_root
+            / self._build_output_path
+            / f"{contract_name}.json",
+            mode="r",
+            encoding="utf-8",
+        ) as compiled_contract_file:
+
+            network_config = self._get_network_config(
+                gateway_url=gateway_url, network=network
+            )
+
+            await deploy(
+                gateway_url=network_config.gateway_url,
+                compiled_contract_file=compiled_contract_file,
+                constructor_args=inputs,
+                salt=salt,
+                token=token,
+            )
+
+    def _get_network_config(
+        self,
+        gateway_url: Optional[str] = None,
+        network: Optional[str] = None,
+    ) -> NetworkConfig:
+        network_config: Optional[NetworkConfig] = None
+
+        if network:
+            network_config = NetworkConfig.from_starknet_network_name(network)
+        if gateway_url:
+            network_config = NetworkConfig(gateway_url=gateway_url)
+
+        if network_config is None:
```

Comment:
Would be cleaner to check it at the beginning of the function

Reply:
I think you confused `network` with `network_config`.

Reply:
true